code/src/components/products-container/ProductsContainer.tsx

```diff
@@ -36,8 +45,18 @@ const ProductsContainer = ({
 
   const skeletonCards = repeatComponent(<ProductSkeleton />, loadingItemsCount);
 
-  const handleAddToCart = () => {
-    openDrawer(<CartDrawer />);
+  const handleAddToCart = async (product: Product) => {
+    try {
+      if (user?.id) {
+        await addToCart({ productId: product.id, userId: user.id }).unwrap();
+        openDrawer(<CartDrawer />);
+      }
+    } catch (e) {
+      openSnackbar({
+        variant: "error",
+        messageTranslationKey: "cart.itemAddition.fail"
+      });
```

Comment:
Do we have conflict with global error snackbar?

Reply:
As far as I see from state changes, global error snackbar handles error and sets according state. Then immediately my snackbar state is setted. Because of transition and instant setting of my snackbar state, message from global snackbar is not being seen

Reply:
Interseting solution!